searchlib/src/vespa/searchlib/query/query_term_ucs4.h

```diff
@@ -14,29 +14,30 @@ namespace search {
  */
 class QueryTermUCS4 : public QueryTermSimple {
 public:
-    typedef std::vector<ucs4_t> UCS4StringT;
     typedef std::unique_ptr<QueryTermUCS4> UP;
-    QueryTermUCS4(const QueryTermUCS4 &) = default;
-    QueryTermUCS4 & operator = (const QueryTermUCS4 &) = default;
+    QueryTermUCS4(const QueryTermUCS4 &) = delete;
+    QueryTermUCS4 & operator = (const QueryTermUCS4 &) = delete;
     QueryTermUCS4(QueryTermUCS4 &&) = default;
     QueryTermUCS4 & operator = (QueryTermUCS4 &&) = default;
     QueryTermUCS4();
     QueryTermUCS4(const string & term_, Type type);
     ~QueryTermUCS4();
-    size_t getTermLen() const { return _cachedTermLen; }
-    size_t term(const char * & t)     const { t = getTerm(); return _cachedTermLen; }
-    UCS4StringT getUCS4Term() const;
+    uint32_t getTermLen() const { return _cachedTermLen; }
+    uint32_t term(const char * & t)     const { t = getTerm(); return _cachedTermLen; }
     void visitMembers(vespalib::ObjectVisitor &visitor) const override;
-    size_t term(const ucs4_t * & t) {
-        if (_termUCS4.empty()) {
-            _termUCS4 = getUCS4Term();
+    uint32_t term(const ucs4_t * & t) {
+        if (!_filled) {
+            fillUCS4();
         }
```

Comment:
Might need a read barrier ?

Reply:
I assume the std:.atomic<bool> does the trick.